spyderlib/widgets/sourcecode/base.py

```diff
@@ -155,6 +157,18 @@ def keyPressEvent(self, event):
         if (key in (Qt.Key_Return, Qt.Key_Enter) and self.enter_select) \
            or key == Qt.Key_Tab:
             self.item_selected()
+            if self.type_list[self.currentRow()] not in ['class', 'function',
+                                                          'method']:
+                return
+            edit = self.textedit
+            s_trailing_text = edit.get_text('cursor', 'eol').strip()
+            if edit.close_parentheses_enabled and \
+              (len(s_trailing_text) == 0 or \
+              s_trailing_text[0] in (',', ')', ']', '}')):

```

Comment:
One space to left here please, so `s_trailing_text` is placed exactly below `len`.

Also, why this second condition? Could you give some examples of when it applies? I mean, why are there here `]` and `}`?